client/state/themes/selectors.js

```diff
@@ -574,6 +574,18 @@ export function isThemePremium( state, themeId ) {
 }
 
 /**
+ * Whether a WPCOM theme given by its ID belongs to the Premium Squared bundle.
+ *
+ * @param  {Object} state   Global state tree
+ * @param  {Object} themeId Theme ID
+ * @return {Boolean}        True if the theme is in the Premium Squared bundle
+ */
+export function isPremiumSquaredTheme( state, themeId ) {
+	const theme = getTheme( state, 'wpcom', themeId );
+	return theme && includes( PREMIUM_SQUARED_THEMES, theme.stylesheet );
```

Comment:
Have you considered using only `themeId` here for checking (i.e. without the `premium/` prefix)? That way, you wouldn't even have to call `getTheme` -- in fact, we wouldn't even need `state` and could turn this selector into a plain util (which should then be moved to `utils.js`).

Alternatively, as we discussed before, instead of a hard-wired list of theme slugs (or IDs), you could also consider checking if `theme.author` was either `Automattic` or `Woo` (that would mean you'd have to keep `getTheme` and `state` tho).